app/src/hooks/report/useReports.ts

```diff
@@ -2,10 +2,14 @@ import { IReport } from '@/models';
 import { isUndefined, useEffectAsync } from '@/utils';
 import { useCallback, useState } from 'react';
 import { useReportClient } from './useReportClient';
+import { GridFilterItem } from '@mui/x-data-grid';
+import { ReportFields } from '@/models';
 
 export type UseReportsResult = {
     /** List of reports */
-    reports: IReport[] | undefined;
+    /** NOTE: reports should NOT be undefined.
+     * I'm feeding the reports to MUI Datagrid directly and it does not take undefined values, empty array is good*/
+    reports: IReport[];
```

Comment:
This is the changes I had to make so I don't have the `cannot read length of undefined` that MUI Datagrid throws because its `rows` prop receive `undefined reports`.
All of this happened because I moved the useReports() from `list/index.js`  to `Table.tsx`